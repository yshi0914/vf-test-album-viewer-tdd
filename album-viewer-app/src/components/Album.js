import React from "react";
import { Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";

const Album = (props) => {
  const { albumData } = props;
  const { name } = props;
  return (
    <li className="list-group-item list-group-item-action flex-column align-items-start text-left">
      <Link to={`/photo/${albumData.id}?albumName=${albumData.title}`}>
        {albumData.id}. {albumData.title} | Name: {name}
      </Link>
    </li>
  );
};

export default Album;
