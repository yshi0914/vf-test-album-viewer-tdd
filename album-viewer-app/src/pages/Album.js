import React, { useState } from "react";
import AlbumList from "../components/AlbumList";

const Album = () => {
  const [albums, setAlbums] = useState([]);
  const [users, setUsers] = useState([]);
  let list = <div>Loading data....</div>;
  fetch(`https://jsonplaceholder.typicode.com/albums`)
    .then((response) => response.json())
    .then((json) => {
      setAlbums(json);
    });
  fetch(`https://jsonplaceholder.typicode.com/users`)
    .then((response) => response.json())
    .then((json) => {
      setUsers(json);
    });

  if (albums.length === 0) {
    list = <h4>Waiting for the data...</h4>;
  }
  return (
    <div>
      <h1>All albums with its associated User Name</h1>
      <AlbumList albumsData={albums} usersData={users} />
      {list}
    </div>
  );
};
export default Album;
