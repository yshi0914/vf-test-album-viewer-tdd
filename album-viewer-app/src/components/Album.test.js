import React from "react";
import Album from "./Album";
import { mount } from "enzyme";
import albumsData from "./albumsTest.json";
import {
  BrowserRouter as Router,
  MemoryRouter,
  Route,
  Switch,
} from "react-router-dom";

// check to see whether the user name prop is passed to component and displayed properly
describe("Name props should be passed to component correctly", () => {
  let wrapper = mount(
    <Router>
      <Album key="1" albumData={albumsData[0]} name="Uncle Sam" />
    </Router>
  );
  test("render the big title of this Album Viewer App", () => {
    expect(wrapper.contains("Uncle Sam")).toEqual(true);
  });
});
// check to see whether the album name is passed to component and displayed properly
describe("Name props should be passed to component correctly", () => {
  let wrapper = mount(
    <Router>
      <Album key="2" albumData={albumsData[0]} name="Uncle Sam" />
    </Router>
  );
  test("render the big title of this Album Viewer App", () => {
    expect(wrapper.contains("quidem molestiae enim")).toEqual(true);
  });
});
// Album component should generate only one Album tag
describe("The size of album should be 100 based on the Album data file", () => {
  let wrapper = mount(
    <Router>
      <Album key="3" albumData={albumsData[1]} name="Uncle Tom" />
    </Router>
  );
  test("render the big title of this Album Viewer App", () => {
    expect(wrapper.find("Album")).toHaveLength(1);
  });
});
// Snapshot Testing for Album component
describe("Album component snapshot testing", () => {
  test("matches the snapshot", () => {
    const component = mount(
      <Router>
        <Album key="3" albumData={albumsData[1]} name="Uncle Tom" />
      </Router>
    );
    expect(component.getElements()).toMatchSnapshot();
  });
});

// Click one of albums, the page should be redirected to the photo page
describe("Click the album, check the behaviour of the page", () => {
  const MockComp = () => <div className="target">Target</div>;
  const MockDenied = () => <div className="denied"> Denied</div>;

  const wrapper = mount(
    <MemoryRouter initialEntries={["/"]}>
      <div>
        <Album key="3" albumData={albumsData[1]} name="Uncle Tom" />
        <Switch>
          <Route exact path="/" component={MockDenied} />
          <Route exact path="/photo/:id" component={MockComp} />
        </Switch>
      </div>
    </MemoryRouter>
  );
  test("Click the album, it should redirect to photo page", () => {
    wrapper.find("a").simulate("click", { button: 0 });
    expect(wrapper.find(".target")).toHaveLength(1);
  });
});
