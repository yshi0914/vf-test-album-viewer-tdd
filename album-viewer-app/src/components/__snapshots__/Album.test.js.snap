// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component snapshot testing matches the snapshot 1`] = `
Array [
  <BrowserRouter>
    <Album
      albumData={
        Object {
          "id": 2,
          "title": "sunt qui excepturi placeat culpa",
          "userId": 1,
        }
      }
      name="Uncle Tom"
    />
  </BrowserRouter>,
]
`;
